phpBB/phpbb/controller/helper.php

```diff
@@ -143,6 +144,15 @@ public function route($route, array $params = array(), $is_amp = true, $session_
 		$context = new RequestContext();
 		$context->fromRequest($this->symfony_request);
 
+		if ($this->config['force_server_vars'])
+		{
+			$context->setHost($this->config['server_name']);
+			$context->setScheme(substr($this->config['server_protocol'], 0, -3));
+			$context->setHttpPort($this->config['server_port']);
+			$context->setHttpsPort($this->config['server_port']);
+			$context->setBaseUrl(rtrim($this->config['script_path'], '/'));
```

Comment:
Yes (if you are generating a full UR and not only a relative path) but that's exactly the purpose of `force_server_vars` and the current applied behaviour in generate_board_url()